### PR TITLE
FEATURE: Hierarchical asset collections

### PR DESCRIPTION
**Pull request by Sebobo**
Adds hierarchical collections functionality

Resolves: #64 #179 #180

### CODE REVIEW

**Comment by bwaidelich** [comment]
Related Neos.Media change: https://github.com/neos/neos-development-collection/pull/4160

File: Resources/Private/JavaScript/asset-collections/src/fragments/assetCollection.ts [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
         id
         title
+        parent {
+            id
+            title
+        }
+        children {
+            id
+            title
+        }
```
Comment (bwaidelich) [inline Resources/Private/JavaScript/asset-collections/src/fragments/assetCollection.ts:15]: Always fetching parent and all children makes this query really really expensive. One of the two should be enough per usecase.
Similar thing for `tags` and `assetCount` do we really need to fetch those for every asset all the time?

**Comment by batabana** [comment]
Hi @Sebobo, we are currently trying to get this feature to run in our project but have not yet been successful. Is there anything "special" we need to do? What we tried is:

- require/reference the feature branch in composer.json
- run doctrine migrations
- check out the media ui in the neos backend -> was not fully functional, especially nesting of asset collections was not possible

From what we understood we should be able to run this feature without the neos core change, correct? Are we missing anything else?

Thanks in advance :)

**Comment by batabana** [comment]
that helped, thank you!

**Comment by Sebobo** [comment]
@batabana regarding the image selection issue, do you have any constraints or media types configured for that nodetypes image property? I'm not sure how to reproduce this problem right now.

**Pull request merged**